source/blog/2017-08-04-ovirt-webadmin-gwt-debug-quick-refresh.html.md

```diff
@@ -0,0 +1,132 @@
+
+## Demo (40 seconds)
+
+<a href="http://www.youtube.com/watch?feature=player_embedded&v=IRGxIrwfnlo
+    " target="_blank"><img src="http://img.youtube.com/vi/IRGxIrwfnlo/0.jpg"
+    alt="demo" width="480" height="360" border="10" /></a>
+
+## Steps
+
+1. Open a terminal, build the engine normally, and start it.
+
+    ```
+    make clean install-dev PREFIX=$HOME/ovirt-engine DEV_EXTRA_BUILD_FLAGS_GWT_DEFAULTS="-Dgwt.cssResourceStyle=pretty -Dgwt.userAgent=safari" BUILD_UT=0 DEV_EXTRA_BUILD_FLAGS="-Dgwt.compiler.localWorkers=1"
+
+    ...
+
+    $HOME/ovirt-engine/share/ovirt-engine/services/ovirt-engine/ovirt-engine.py start
+
+    ```
+
+    ![screen](../images/blog/2017-08-04/1.png)
+
+    ![screen](../images/blog/2017-08-04/2.png)
+
+    ![screen](../images/blog/2017-08-04/3.png)
+
+2. In a second terminal, run:
+
+    Chrome:
+
+    ```
+    make gwt-debug DEV_BUILD_GWT_SUPER_DEV_MODE=1 DEV_EXTRA_BUILD_FLAGS_GWT_DEFAULTS="-Dgwt.userAgent=safari"
+    ```
+
+    or
+
+    Firefox:
+
+    ```
+    make gwt-debug DEV_BUILD_GWT_SUPER_DEV_MODE=1 DEV_EXTRA_BUILD_FLAGS_GWT_DEFAULTS="-Dgwt.userAgent=gecko1_8"
+    ```
+
+    ![screen](../images/blog/2017-08-04/4.png)
+
+    wait about 2 mins for it to say "code server running at http://<<server>>:9876/", and the GWT Dev Mode app will open.
+
+    ![screen](../images/blog/2017-08-04/5.png)
+
+3.  Open your browser and go to http://localhost:9876/ **You must use localhost. it breaks with IPs, not sure why**
+
+    Now you'll see two buttons, Dev Mode On and Dev Mode Off.
+
+    ![screen](../images/blog/2017-08-04/6.png)
+
+    Drag those buttons to the browser bookmark bar, as it says on that page.
+
+    ![screen](../images/blog/2017-08-04/7.png)
+
+4. **Caveat** -- ignore the link under the two buttons. Instead, in a second tab, go to webadmin via normal url, http://localhost:8080
+
+    ![screen](../images/blog/2017-08-04/8.png)
+
+5. Log in and navigate to admin portal
+
+6. [Chrome only: Hit F12, hamburger menu, Settings. Disable both JavaScript and CSS source maps. Our app is too large and the source maps crash Chrome.]
+
+    ![screen](../images/blog/2017-08-04/9.png)
+
+    ![screen](../images/blog/2017-08-04/10.png)
+
+    ![screen](../images/blog/2017-08-04/11.png)
+
+7. click Dev Mode On
+
+    ![screen](../images/blog/2017-08-04/12.png)
+
+8. drag the Compile button to your bookmarks bar
+
+    ![screen](../images/blog/2017-08-04/13.png)
+
+9. click Compile
+
+    ![screen](../images/blog/2017-08-04/14.png)
+
+    this first time is slow, up to 2 minutes
+
+10. make a change to GWT code. Here I'm creating a new "About" dialog box that has an image as a background. I'll tweak the font properties of the dialog's close button to demonstrate the refresh.
```

Comment:
Suggest:
"10. Make a change to the GWT code. Here I am creating a new "About" dialog box that has an image as a background. I will tweak the font properties of the dialog's Close button to demonstrate the refresh."